var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -50,12 +66,67 @@ class Mesa(AutotoolsPackage):
     depends_on('libxext')
     depends_on('libxdamage')
     depends_on('libxfixes')
+    depends_on('expat')
 
     depends_on('glproto@1.4.14:', type='build')
     depends_on('dri2proto@2.6:', type='build')
     depends_on('dri3proto@1.0:', type='build')
     depends_on('presentproto@1.0:', type='build')
     depends_on('pkg-config@0.9.0:', type='build')
+    depends_on('binutils', type='build')
 
     # TODO: Add package for systemd, provides libudev
     # Using the system package manager to install systemd didn't work for me
+
+    def url_for_version(self, version):
+        """Handle mesa version-based custom URLs."""
+        latest_mver = '11'
+
+        if version < Version(latest_mver):
+            base = Mesa.base_url + "/older-versions/%s.x" % (version.up_to(1))
+        else:
+            base = Mesa.base_url
+        if version > Version('10.4'):
+            name = 'mesa'
+        else:
+            name = 'MesaLib'
+        return base + "/%s/%s-%s.tar.gz" % (version, name, version)
+
+    def check_variants(self, var='', ver='', ref=''):
```

Comment:
please use `conflicts('+galium', when='@:9.2.2')` statements as this is done during **conretization** instead of during the build.